### DISCUSSION TITLE
Add a --preview option to upgrade/downgrade commands to list the migrations that will be applied

### DISCUSSION THREAD

**Discussion by eliasdorneles** [sqlalchemy/alembic#1016] [Ideas]
What do you think about adding a `--preview` option to the commands `alembic upgrade` and `alembic downgrade` to list the migrations that would be applied?

Right now, if I want to know what will happen if I run `alembic upgrade`, I have to run `alembic history -i` and "calculate" it in my head.

It would be super handy to have a `--preview` option.

**Comment by zzzeek** [comment]
`--sql` option?  you mean you just want to show the revision identifiers.  You can do "alembic history -r current:head" right now, hows that?

**Reply by zzzeek** [reply]
https://alembic.sqlalchemy.org/en/latest/tutorial.html?highlight=range#viewing-history-ranges

**Reply by eliasdorneles** [reply]
Ahh yes, thank you, `alembic history -r current:head` does what i want!!
Okay, I think I really need to give it a closer read at the documentation! :sweat_smile:
